crates/swc_ecma_transforms_react/src/jsx/mod.rs

```diff
@@ -136,7 +136,7 @@ pub fn parse_expr_for_jsx(
     name: &str,
     src: Lrc<String>,
     top_level_mark: Mark,
-) -> Lrc<Box<Expr>> {
+) -> Box<Expr> {
```

Comment:
Sidenote: this actually probably should just return an Expr then someone could put it in an Lrc or Box after?

Reply:
It seems that the codebase too often pre-emptively boxes stuff. For example, `parse_file_as_expr` should probably return an `Expr` instead of `Box<Expr>` and then this could return an `Expr` and then the caller could decide to put this in an Lrc or Box. This would be a big change though.

Reply:
Returning `Box<Expr>` is better if you can't be 100% sure that `rustc` will see `box some_expression()` for all instance of `some_expression()`

Reply:
Did you profile the performance using a profiler? You will notice that it becomes slower without boxing, although the situation is a bit different from the time I made decisions.